docs/bin/dump_config.py

```diff
@@ -62,11 +63,10 @@ def main(args):
     output_name = os.path.join(output_dir, template_file.replace('.j2', ''))
     temp_vars = {'config_options': config_options}
 
-    with open(output_name, 'wb') as f:
-        f.write(template.render(temp_vars).encode('utf-8'))
+    data = template.render(temp_vars).encode('utf-8')
 
-    return 0
+    update_file_if_different(output_name, data)
 
 
 if __name__ == '__main__':
-    sys.exit(main(sys.argv[:]))
+    main(sys.argv[:])
```

Comment:
Passing `sys.argv[:]` here is useless wrt current implementation.
If you look at the `main()`'s first three lines there's:
```python
def main(args):
    ...
    (options, args) = parser.parse_args()  # <-- immediatelly rewrites args variable
```

It looks like the initial implementation has been written by the person with some C-like background, where main accepts args data and returns 0. But in fact in this case it's not needed.

Please remove this arg.

Reply:
@webknjaz You are correct, and this code, as well as the other scripts are not beauties. But I do not intend to rewrite complete scripts. Can you maybe propose this in a subsequent PR once this one is merged ? A cleanup would be very welcome indeed.

Reply:
Oops, I missed your comment and already did that in commit...

Reply:
Anyway, removal of `return 0` was not related to the purpose of the PR, but related to my refactoring suggestion. So it fits next to your change :)

Reply:
@webknjaz Yeah, but if you continue in that line, you end up rewriting all the scripts and that is not something your colleagues would accept anyway.

Reply:
Agreed

Reply:
I agree with dag that we don't want to change args handling in this PR but for the record the correct change here is to use args in the call to parse_args rather than removing sys.argv from the call to main():

```
def main(args):
    ...
    (options, args) = parser.parse_args(args)
[...]

if __name__ == '__main__':
    sys.exit(main(sys.argv[1:]))
```

Code that follows this style is easier to test and easier to reason about.